src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.Node.cs

```diff
@@ -45,9 +53,11 @@ public static void ThrowNotSupportedException_CollectionIsReadOnly()
         }
 
         [DoesNotReturn]
-        public static void ThrowInvalidOperationException_NodeWrongType(string typeName)
+        public static void ThrowInvalidOperationException_NodeWrongType(params ReadOnlySpan<string> supportedTypeNames)
         {
-            throw new InvalidOperationException(SR.Format(SR.NodeWrongType, typeName));
+            Debug.Assert(supportedTypeNames.Length > 0);
+            string concatenatedNames = supportedTypeNames.Length == 1 ? supportedTypeNames[0] : string.Join(", ", supportedTypeNames.ToArray());
```

Comment:
.NET 9 has an overload of Join that would work without the ToArray, but maybe it's not worth the ifdef for an error path?

Reply:
I don't think it's worth it, particularly given that it's an exceptional code path.